grails-app/controllers/au/org/emii/aatams/AbstractController.groovy

```diff
@@ -22,11 +22,14 @@ class AbstractController
         flattenParams()
 
         def clazz = reportInfoService.getClassForName(queryName)
-        if (resultList.count < grailsApplication.config.filter.count.max + 1) {
-            flash.message = "${resultList.count} matching records (${clazz.count()} total)."
-        }
-        else {
-            flash.message = "&gt; ${grailsApplication.config.filter.count.max} matching records (${clazz.count()} total)."
+
+        if (queryName != "detection") {
```

Comment:
This  code in in the AbstractController which is the superclass for a whole number of controllers. It would be preferable if we could avoid any special cases in here. One of the subclasses is a DetectionController which would be the perfect place for some detection-specific behaviour.

If we override a method in the subclass it'd be best to minimise the amount of duplication. So I think it'd be too much to override and reimplement all of `doList()`. Perhaps we just extract the part which sets the count message into a new method. Check it out and see what you can see. Let me know if you have any questions.